.github/workflows/demo-more-modules.yaml

```
@@ -0,0 +1,33 @@
+name: Build more-modules app
+
+on:
+  workflow_dispatch: { }
+  push: { }
+  pull_request: { }
```

GitHub reports that four workflows have run, where I would've expected only two. I guess this is because this workflow (and the other) trigger on a push to _any_ branch as well as on pull requests.

```suggestion
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```

I think the above suggestion should address that. It lets the workflow trigger on pushes to the **main** branch or pull requests against the **main** branch, if I'm not mistaken.